src/hooks/useApi.js

```diff
@@ -0,0 +1,64 @@
+import axios from "axios";
+import { useCallback } from "react";
+import useSWR from "swr";
+
+/**
+ * @param {string} url - excluding base, i.e. /api/v1/users` should just be `/users`.
+ * @param {import("axios").AxiosRequestConfig} options - axios options.
+ * @returns
+ */
+export const useApi = (url, options) => {
+  const { data, error, ...rest } = useSWR(getUrl(url), fetcher(options));
+  const loading = !data && !error;
+
+  return { data, error, ...rest, loading };
+};
+
+/**
+ * @param {string} url
+ * @param {import("axios").AxiosRequestConfig} options
+ *
+ */
+export const useMutation = (url, options) => {
+  const fn = useCallback(
+    /**
+     * @param {import("axios").AxiosRequestConfig} overrideOptions
+     */
+    (overrideOptions, newUrl = url) => {
+      return fetcher({
+        ...{ ...options, ...overrideOptions },
+      })(getUrl(newUrl));
+    },
+    [options, url]
+  );
+
+  return fn;
+};
+
+/**
+ * @param { import("axios").AxiosRequestConfig } axiosConfig
+ */
+export const fetcher = (axiosConfig) => {
+  /** @param {string} url */
+  return async (url) => {
+    const request = await axios(url, axiosConfig);
+    return request.data;
+  };
+};
+
+/**
+ * @param {string} url
+ * @returns {string}
+ */
+export const getUrl = (url) => {
+  if (!url.startsWith("/")) {
+    url = "/" + url;
+  }
+
+  const base =
+    process.env.NODE_ENV === "development"
```

Comment:
I remembered there is a similar line in src/api/index.js,
did that config anything? if not should we just remove it?